pinot-core/src/main/java/org/apache/pinot/core/operator/transform/function/BaseTransformFunction.java

```diff
@@ -588,4 +589,34 @@ public String[][] transformToStringValuesMV(ProjectionBlock projectionBlock) {
     }
     return _stringValuesMV;
   }
+
+  @Override
+  public byte[][][] transformToBytesValuesMV(ProjectionBlock projectionBlock) {
+    int length = projectionBlock.getNumDocs();
+    if (_bytesValuesMV == null) {
+      _bytesValuesMV = new byte[length][][];
+    }
+    Dictionary dictionary = getDictionary();
+    if (dictionary != null) {
+      int[][] dictIdsMV = transformToDictIdsMV(projectionBlock);
+      for (int i = 0; i < length; i++) {
+        int[] dictIds = dictIdsMV[i];
+        int numValues = dictIds.length;
+        byte[][] bytesValues = new byte[numValues][];
+        dictionary.readBytesValues(dictIds, numValues, bytesValues);
+        _bytesValuesMV[i] = bytesValues;
+      }
+    } else {
+      Preconditions.checkState(getResultMetadata().getDataType().getStoredType() == DataType.STRING);
```

Comment:
Use `assert` instead which doesn't have overhead
```suggestion
      assert getResultMetadata().getDataType().getStoredType() == DataType.STRING;
```